plugin/core/views.py

```diff
@@ -176,20 +176,58 @@ def text_document_range_formatting(view: sublime.View, region: sublime.Region) -
     })
 
 
-def minihtml(view: sublime.View, content: Union[str, dict]) -> str:
-    """ Content can be a string or MarkupContent. """
+def minihtml(view: sublime.View, content: Union[str, dict, list], prefer_plain_text: bool) -> str:
+    """
+    Formats provided input into markup accepted by minihtml.
+
+    Content can be in one of those formats:
+     - string: treated as plain text
+     - MarkedString: string | { language: string; value: string }
+     - MarkedString[]
+     - MarkupContent: { kind: MarkupKind, value: string }
+
+    Since we can't distinguish between plain text string and a MarkedString in a string form, we
+    need to be told explicitly which one to use for string input.
+
+    :param view
+    :param content
+    :param prefer_plain_text: Whether to treat *string* input as plain text (True) or markdown (False).
+
+    :returns: Formatted string
+    """
     if isinstance(content, str):
-        return text2html(content)
-    elif isinstance(content, dict):
-        value = content.get("value") or ""
+        return text2html(content) if prefer_plain_text else mdpopups.md2html(view, content)
+    if isinstance(content, list):
+        formatted = []
+        for item in content:
+            value = ""
+            language = None
+            if isinstance(item, str):
+                value = item
+            else:
+                value = item.get("value") or ""
+                language = item.get("language")
+
+            if language:
+                formatted.append("```{}\n{}\n```\n".format(language, value))
+            else:
+                formatted.append(value)
+
+        frontmatter_config = mdpopups.format_frontmatter({'allow_code_wrap': True})
+        return mdpopups.md2html(view, frontmatter_config + "\n".join(formatted))
+    if isinstance(content, dict):
+        language = content.get("language")
```

Comment:
I really like how you added comments like `# MarkupContent` or `# MarkupContent` after each if statement.
can you add them after all ifs, it really useful.(I don't have to look it up or lose 1min tring to figure out if it is a MarkedString or MarkupKind or a list of MarkedStrings)